pkg/checkpoint/pod.go

```diff
@@ -72,6 +72,15 @@ func sanitizeCheckpointPod(cp *v1.Pod) *v1.Pod {
 		} else if v.ConfigMap != nil {
 			v.HostPath = &v1.HostPathVolumeSource{Path: configMapPath(cp.Namespace, cp.Name, v.ConfigMap.Name)}
 			v.ConfigMap = nil
+		} else if v.Projected != nil {
+			// can only currently support projected volumes with all secrets or all configmaps
+			if volumeHasSecretProjection(v) {
+				v.HostPath = &v1.HostPathVolumeSource{Path: secretPath(cp.Namespace, cp.Name, v.Name)}
+				v.Projected = nil
+			} else if volumeHasConfigMapProjection(v) {
```

Comment:
a projection could have a secret and a configmap, right? So this else if would be just an if statement.

Reply:
please see update to pull request description

Reply:
it's not something my work colleagues and I currently need but we would like some basic support for projected volumes hence the PR :)

Reply:
I would expect this to support a projected volume with both configmaps and secrets. It doesn't make sense to advertise this as "supporting projected volumes" otherwise.

Reply:
OK, so I need to serialise secrets and config maps into the same directory for a volume.

Would you prefer all secrets and config maps had the same base directory which may break existing users, or should I create a 3rd base directory for secrets and config maps from projected volumes?

Reply:
@ericchiang Any update on this?

Reply:
@ericchiang sorry to bump this again, have you been able to take a look at this suggested solution?

Reply:
@ericchiang @rphillips Is there anything to be done to progress on this PR?

Reply:
Is there a reason why this code path is restricted to only all secrets or all configmaps? It feels like we can make this less error prone and support any combination of projections.

I worry about the edge cases and currently running deployments breaking. Adding a flag to enable projection (`--with-projection`) would mitigate the concern so we can test in current deployments. 

/cc @aaronlevy thoughts?

Reply:
I would lean toward not adding flags that changes behavior. If we can't reason about safety - adding a flag is just going to make reasoning about safety of the behavior even more difficult in the future.

Reply:
@rphillips we could create a separate base directory to store projected volume files and would have made that change had @ericchiang not made a comment that they needed to think about it a little

Reply:
@rphillips @aaronlevy Do you have any further thoughts on this?

Reply:
Can we add tests for all projected volume types (as well as mixed types within same pod) to demonstrate that this has been resolved?

Reply:
Also was cleanup of the checkpointed data handled?